lib/ukstore/include/uk/store.h

```diff
@@ -29,3 +29,404 @@
  * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
  * POSSIBILITY OF SUCH DAMAGE.
  */
+
+#ifndef __UK_STORE_H__
+#define __UK_STORE_H__
+
+#include <string.h>
+#include <uk/config.h>
+#include <uk/assert.h>
+#include <uk/list.h>
+#include <uk/arch/atomic.h>
+#include <uk/essentials.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#if !CONFIG_LIBUKSTORE
+
+#define uk_store_libid(libname) (-1)
+
+#define uk_store_dynamic_entry_create(folder, name, type, get, set)	\
```

Comment:
True. Though, I wouldn't do the dynamic/static separation. In the last fixes we made the access to static/dynamic entries seamless. It would be strange to separate them again. Also, i think it gets cumbersome to use both `CONFIG_LIBUKSTORE` and `CONFIG_LIBUKSTORE_DYNAMIC`.

The other suggestions are all good.